#!/usr/bin/env python
PACKAGE = "aiv_controller_pid"

from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()

gen.add("r1_kp", double_t, 0, "kp value r1", 1, 0, 10)
gen.add("r1_ki", double_t, 0, "ki value r1", 1, 0, 10)
gen.add("r1_kd", double_t, 0, "kd value r1", 1, 0, 10)
gen.add("l2_kp", double_t, 0, "kp value l2", 1, 0, 10)
gen.add("l2_ki", double_t, 0, "ki value l2", 1, 0, 10)
gen.add("l2_kd", double_t, 0, "kd value l2", 1, 0, 10)
gen.add("l3_kp", double_t, 0, "kp value l3", 1, 0, 10)
gen.add("l3_ki", double_t, 0, "ki value l3", 1, 0, 10)
gen.add("l3_kd", double_t, 0, "kd value l3", 1, 0, 10)
gen.add("r4_kp", double_t, 0, "kp value r4", 1, 0, 10)
gen.add("r4_ki", double_t, 0, "ki value r4", 1, 0, 10)
gen.add("r4_kd", double_t, 0, "kd value r4", 1, 0, 10)
gen.add("x_kp", double_t, 0, "kp value odo_x", 1, 0, 10)
gen.add("x_ki", double_t, 0, "ki value odo_x", 1, 0, 10)
gen.add("x_kd", double_t, 0, "kd value odo_x", 1, 0, 10)
gen.add("y_kp", double_t, 0, "kp value odo_y", 1, 0, 10)
gen.add("y_ki", double_t, 0, "ki value odo_y", 1, 0, 10)
gen.add("y_kd", double_t, 0, "kd value odo_y", 1, 0, 10)
gen.add("th_kp", double_t, 0, "kp value odo_th", 1, 0, 10)
gen.add("th_ki", double_t, 0, "ki value odo_th", 1, 0, 10)
gen.add("th_kd", double_t, 0, "kd value odo_th", 1, 0, 10)

exit(gen.generate(PACKAGE, "aiv_controller_pid", "controllerServer"))
